interpreter/syntax/types.ml

```diff
@@ -1,11 +1,23 @@
 (* Types *)
 
-type num_type = I32Type | I64Type | F32Type | F64Type
-type vec_type = V128Type
-type ref_type = FuncRefType | ExternRefType
-type value_type = NumType of num_type | VecType of vec_type | RefType of ref_type
-type result_type = value_type list
-type func_type = FuncType of result_type * result_type
+type name = int list
+
+and syn_var = int32
+and sem_var = def_type Lib.Promise.t
+and var = SynVar of syn_var | SemVar of sem_var
+
+and nullability = NonNullable | Nullable
+and num_type = I32Type | I64Type | F32Type | F64Type
+and vec_type = V128Type
+and ref_type = nullability * heap_type
+and heap_type =
+  FuncHeapType | ExternHeapType | DefHeapType of var | BotHeapType
+and value_type =
+  NumType of num_type | VecType of vec_type | RefType of ref_type | BotType
```

Comment:
What is a BotType?

Reply:
The bottom type. It only occurs internally.